proxy/http2/Http2Stream.cc

```diff
@@ -767,8 +763,14 @@ Http2Stream::destroy()
     h2_proxy_ssn->connection_state.release_stream(this);
 
     ssn_id = _proxy_ssn->get_id();
+
+    if (_proxy_ssn->get_netvc()->get_context() == NET_VCONNECTION_IN) {
+      HTTP2_DECREMENT_THREAD_DYN_STAT(HTTP2_STAT_CURRENT_CLIENT_STREAM_COUNT, _thread);
```

Comment:
Use of destructors sounds great.

This is just an input for following changes. It would be great if we could generalize the destruction process, and put the code into ProxyTransaction's destructor and the details like this line into subclasses. This is the reason I was talking about `decrement_txn_stat` on a previous PR.